.github/workflows/build.yml

```diff
@@ -0,0 +1,195 @@
+  JOBS: 2
+  DEBUG: 1
+  # Useful URL: https://github.com/mate-desktop/debian-packages
+  # Useful URL: https://salsa.debian.org/debian-mate-team/mate-panel
+  DEB_LIBRARY_DEPS: |
+    libatk1.0-dev
+    libcairo2-dev
+    libdconf-dev
+    libgirepository1.0-dev
+    libglib2.0-dev
+    libgtk-3-dev
+    libgtk-layer-shell-dev
+    libice-dev
+    libmate-desktop-dev
+    libmate-menu-dev
+    libmateweather-dev
+    libpango1.0-dev
+    libsm-dev
+    libsoup2.4-dev
+    libwnck-3-dev
+    libx11-dev
+    libxrandr-dev
+    libgirepository1.0-dev
+  DEB_BUILD_DEPS: |
+    ccache
+    autoconf-archive
+    autopoint
+    gir1.2-freedesktop
+    git
+    gobject-introspection
+    gtk-doc-tools
+    lsb-release
+    make
+    mate-common
+    meson
+    yelp-tools
+    iso-codes
+    gobject-introspection
+  # TODO
+  DEB_SCAN_BUILD_DEPS: |
+    clang
+    clang-tools
+  # Useful URL: https://git.archlinux.org/svntogit/community.git/tree/mate-panel
+  # + mate-desktop dependencies
+  ARCH_BUILD_DEPS: |
+    ccache
+    autoconf-archive
+    clang
+    gcc
+    git
+    gobject-introspection
+    gtk-layer-shell
+    itstool
+    libcanberra
+    libmateweather
+    libsm
+    libwnck3
+    make
+    mate-common
+    mate-desktop
+    mate-menus
+    meson
+    which
+    yelp-tools
+    iso-codes
+    gobject-introspection
+
+jobs:
+  build:
+    name: Build on ${{matrix.container}} with in-process=${{matrix.in-process}}
+    runs-on: ubuntu-latest
+    container: ${{matrix.container}}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        in-process: [all, none]
+        container: ['debian:testing', 'ubuntu:rolling', 'archlinux:latest']
+
+    env:
+      CC: ccache gcc
+      CXX: ccache g++
+      BASE_INSTALL_PATH: ${{github.workspace}}/_install
+
+    steps:
+      - name: Setup environment
+        run: |
+          echo "PATH=${BASE_INSTALL_PATH}/bin:${PATH}" >> "$GITHUB_ENV"
+          echo "PKG_CONFIG_PATH=${BASE_INSTALL_PATH}/lib/pkgconfig:${PKG_CONFIG_PATH}" >> "$GITHUB_ENV"
+
+      - name: Show environment
+        run: env | sort
+        if: ${{ env.DEBUG == '1' }}
+
+      # For Debian and Ubuntu (apt-based with reasonably compatible packages)
+      - name: Install dependencies
+        if: ${{ startsWith(matrix.container, 'debian:') || startsWith(matrix.container, 'ubuntu:') }}
+        run: |
+          apt-get update -qq
+          apt-get install --assume-yes --no-install-recommends \
+            ${DEB_BUILD_DEPS} ${DEB_LIBRARY_DEPS}
+
+      # For ArchLinux
+      - name: Install dependencies
+        if: ${{ startsWith(matrix.container, 'archlinux:') }}
+        # don't upgrade, although told otherwise (see link below), because
+        # apparently in the container it doesn't quit work...
+        # https://wiki.archlinux.org/title/System_maintenance#Partial_upgrades_are_unsupported
+        run: |
+          pacman --noconfirm -Sy
+          pacman --noconfirm -S ${ARCH_BUILD_DEPS}
```

Comment:
If anybody knows better, we can try again `-Syu` as recommended, but I never used Arch and it seems it has issues in the docker container, and I had no idea what that meant.

Reply:
@zhuyaliang @yetist ^^^^

Reply:
What is the error content

Reply:
It was around there: https://github.com/mate-desktop/mate-panel/actions/runs/6867238053/job/18675135924?pr=1415#step:6:113.  Apparently it works, so either it was a transitive error, or I was doing something else wrong and though this was fatal (possible).
Anyway, it seems to work now, so we can probably resolve this.